src/shared/bootspec.c

```diff
@@ -392,13 +635,23 @@ int boot_entries_load_config(
                 r = boot_entries_find(p, esp_path, &config->entries, &config->n_entries);
                 if (r < 0)
                         return r;
+
+                p = strjoina(esp_path, "/EFI/Linux/");
+                r = boot_entries_find_unified(p, esp_path, &config->entries, &config->n_entries);
+                if (r < 0)
+                        return r;
```

Comment:
Hmm, if any of the entries are invalid (e.g. missing os-release fields), we're fail hard here. I guess it'd be better to warn and ignore invalid entries.